media-sound/pulseaudio/pulseaudio-14.2.ebuild

```diff
@@ -0,0 +1,318 @@
+	udev? ( || ( alsa oss ) )
+	zeroconf? ( dbus )
+"
+
+# libpcre needed in some cases, bug #472228
+RDEPEND="
+	|| (
+		elibc_glibc? ( virtual/libc )
+		elibc_uclibc? ( virtual/libc )
+		dev-libs/libpcre
+	)
+	>=media-libs/libsndfile-1.0.20[${MULTILIB_USEDEP}]
+	X? (
+		>=x11-libs/libX11-1.4.0[${MULTILIB_USEDEP}]
+		>=x11-libs/libxcb-1.6[${MULTILIB_USEDEP}]
+		x11-libs/libSM[${MULTILIB_USEDEP}]
+		x11-libs/libICE[${MULTILIB_USEDEP}]
+		x11-libs/libXtst[${MULTILIB_USEDEP}]
+	)
+	>=sys-libs/libcap-2.22-r2[${MULTILIB_USEDEP}]
+	alsa? ( >=media-libs/alsa-lib-1.0.19 )
+	glib? ( >=dev-libs/glib-2.26.0:2[${MULTILIB_USEDEP}] )
+	zeroconf? ( >=net-dns/avahi-0.6.12[dbus] )
+	jack? ( virtual/jack )
+	tcpd? ( sys-apps/tcp-wrappers[${MULTILIB_USEDEP}] )
+	lirc? ( app-misc/lirc )
+	dbus? ( >=sys-apps/dbus-1.0.0[${MULTILIB_USEDEP}] )
+	gtk? ( x11-libs/gtk+:3 )
+	bluetooth? (
+		>=net-wireless/bluez-5
+		media-libs/sbc
+	)
+	asyncns? ( net-libs/libasyncns[${MULTILIB_USEDEP}] )
+	udev? ( >=virtual/udev-143[hwdb(+)] )
+	equalizer? (
+		sci-libs/fftw:3.0
+	)
+	ofono-headset? ( >=net-misc/ofono-1.13 )
+	orc? ( >=dev-lang/orc-0.4.15 )
+	sox? ( >=media-libs/soxr-0.1.1 )
+	ssl? ( dev-libs/openssl:0= )
+	media-libs/speexdsp
+	gdbm? ( sys-libs/gdbm:= )
+	webrtc-aec? ( >=media-libs/webrtc-audio-processing-0.2 )
+	systemd? ( sys-apps/systemd:0=[${MULTILIB_USEDEP}] )
+	!systemd? ( sys-auth/elogind )
+	dev-libs/libltdl:0
+	selinux? ( sec-policy/selinux-pulseaudio )
+" # libltdl is a valid RDEPEND, libltdl.so is used for native abi in pulsecore and daemon
+
+DEPEND="${RDEPEND}
+	X? (
+		x11-base/xorg-proto
+		>=x11-libs/libXtst-1.0.99.2[${MULTILIB_USEDEP}]
+	)
+	dev-libs/libatomic_ops
+"
+# This is a PDEPEND to avoid a circular dep
+PDEPEND="
+	alsa? ( alsa-plugin? ( >=media-plugins/alsa-plugins-1.0.27-r1[pulseaudio,${MULTILIB_USEDEP}] ) )
+"
+
+# alsa-utils dep is for the alsasound init.d script (see bug #155707); TODO: read it
+RDEPEND="${RDEPEND}
+	system-wide? (
+		alsa? ( media-sound/alsa-utils )
+		acct-user/pulse
+		acct-group/pulse-access
+	)
+	acct-group/audio
+"
+
+BDEPEND="
+	doc? ( app-doc/doxygen )
+	system-wide? ( dev-util/unifdef )
+	test? ( >=dev-libs/check-0.9.10 )
+	sys-devel/gettext
+	sys-devel/m4
+	virtual/pkgconfig
+"
+
+# BUG: work-around for a weird bug
+#DOCS=( NEWS README ) # todo is useless to install
+
+#PATCHES=()
+
+pkg_pretend() {
+	CONFIG_CHECK="~HIGH_RES_TIMERS"
+	WARNING_HIGH_RES_TIMERS="CONFIG_HIGH_RES_TIMERS:\tis not set (required for enabling timer-based scheduling in pulseaudio)\n"
+	check_extra_config
+
+	if linux_config_exists; then
+		local snd_hda_prealloc_size=$(linux_chkconfig_string SND_HDA_PREALLOC_SIZE)
+		if [ -n "${snd_hda_prealloc_size}" ] && [ "${snd_hda_prealloc_size}" -lt 2048 ]; then
+			ewarn "A pre-allocated buffer-size of 2048 (kB) or higher is recommended for the"
+			ewarn "HD-audio driver when using timer-based scheduling!"
+			ewarn "CONFIG_SND_HDA_PREALLOC_SIZE=${snd_hda_prealloc_size}"
```

Comment:
It's a bit complicated - the new default, I think, is to set it to 0 and have no pre-allocation at all. But, as far as I understand, that also means that there's no guarantee the runtime created buffer will be continuous (or something to that extent). It's further complicated by the fact that presumably we'd like to move towards using PipeWire in the future, which does not currently have the ability to fully utilize buffers even close to that size, leading to wasted memory and allegedly CPU cycles (I guess there's some inherent inefficiency when the application buffers are much smaller than the hardware buffer). For now I have opted to be cautious and not change our kernel configuration recommendation but I more than welcome anyone wishing to express their opinion on this matter to do so either in this PR or perhaps more appropriately on bugs.gentoo.org.

Reply:
That sounds fine, although it might be worth mentioning this in a comment and tagging the relevant bug(s) to show we've considered it. Thanks for the thought out answer!

Reply:
I opted to just remove the kernel configuration checks entirely - it's been so long that anyone who cared has long updated their kernel configs and I don't feel like telling people to instead set it to 0 because that would benefit another sound daemon.